src/dense.jl

```diff
@@ -39,46 +39,65 @@ struct AutoDiffractor <: AbstractADType end
 mode(::AutoDiffractor) = ForwardOrReverseMode()
 
 """
-    AutoEnzyme{M,A}
+    AutoEnzyme{M,A,C}
 
 Struct used to select the [Enzyme.jl](https://github.com/EnzymeAD/Enzyme.jl) backend for automatic differentiation.
 
 Defined by [ADTypes.jl](https://github.com/SciML/ADTypes.jl).
 
 # Constructors
 
-    AutoEnzyme(; mode::M=nothing, function_annotation::Type{A}=Nothing)
+    AutoEnzyme(;
+        mode::Union{EnzymeCore.Mode,Nothing}=nothing,
+        function_annotation::Type{<:Union{EnzymeCore.Annotation,Nothing}}=Nothing,
+        chunksize::Union{Int,Float64,Nothing}=nothing,
+    )
 
-# Type parameters
+  - `mode::M` determines the autodiff mode (forward or reverse). It can be:
 
-  - `A` determines how the function `f` to differentiate is passed to Enzyme. It can be:
+      + a mode object from EnzymeCore.jl, like `EnzymeCore.Forward` or `EnzymeCore.Reverse` (possibly modified with additional settings like runtime activity)
+      + `nothing` to choose the best mode automatically
 
-      + a subtype of `EnzymeCore.Annotation` (like `EnzymeCore.Const` or `EnzymeCore.Duplicated`) to enforce a given annotation
-      + `Nothing` to simply pass `f` and let Enzyme choose the most appropriate annotation
+  - `A=function_annotation` determines how the function `f` to differentiate is passed to Enzyme. It can be:
 
-# Fields
+      + a subtype of `EnzymeCore.Annotation` (like `EnzymeCore.Const` or `EnzymeCore.Duplicated`) to enforce a given annotation
 
-  - `mode::M` determines the autodiff mode (forward or reverse). It can be:
+      + `Nothing` (the type, not the object) to simply pass `f` and let Enzyme choose the most appropriate annotation
+  - `C=chunksize` determines the number of derivatives evaluated simultaneously when computing operators like a Jacobian or a forward-mode gradient. It can be:
 
-      + an object subtyping `EnzymeCore.Mode` (like `EnzymeCore.Forward` or `EnzymeCore.Reverse`) if a specific mode is required
-      + `nothing` to choose the best mode automatically
+      + a positive `Int` to fix a constant chunk size
```

Comment:
I kind of wonder if a chunk size of 0 here would be a good way to represent maximum chunk size

Reply:
reading that, I still don't understand why a zero chunksize is semantically meaningful?

Reply:
Hi Billy, just following up on this, do you want to add that setting to EnzymeCore?

Reply:
See https://github.com/EnzymeAD/Enzyme.jl/pull/2659